### PR TITLE
Automation of observation converter creation

### PR DESCRIPTION
**Pull request by c-merchant**
## Description:
This PR provides a script and associated infrastructure to automatically generate an observation converter based on the input data format and gets the user started with development, documentation, and building. The previously generalized "text" observation converter was absorbed into this feature with the associated infrastructure kept largely the same. The readme provides condensed documentation for observation converters specifically and from references across other pages in the guide as well as a new step-by-step guide on how to go about creating an observation converter. The existing documentation has been edited for clarity and flow. 

To do: the toctree for the observation converter documentation displays in the webpage, and there may be a .gitignore floating around somewhere. 
### Fixes issue
fixes: #684 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Documentation changes needed?

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### Tests
I ran the quickbuild files for each data type for the observation converter, worked through a brief implementation based on the available files, and ran the standard ./quickbuild.sh, ./perfect_model_obs, and ./filter for Lorenz-63 to ensure no bugs were introduced.

## Checklist for merging
- [ ] Updated changelog entry
- [ ] Documentation updated
- [ ] Update conf.py

## Checklist for release
- [ ] Merge into main
- [ ] Create release from the main branch with appropriate tag
- [ ] Delete feature-branch

## Testing Datasets
- [ ] Dataset needed for testing available upon request
- [ ] Dataset download instructions included
- [x] No dataset needed

### CODE REVIEW

**Comment by hkershaw-brown** [comment]
Thanks for this Charlotte, it might be a little while before we look at this.

Cheers,
Helen

**Review by Benjamin-Gunn — APPROVED** [review]
Everything looked good to me!

File: observations/obs_converters/README.rst [hunk 1]
```diff
@@ -1,361 +1,45 @@
...
-It is common that an assimilation will want to use observations
-from different sources.  Generally it is easier to convert observations
-from each source separately and then merge them together with the
-``obs_sequence_tool``.
-
-Creating filenames and directory names which follow a pattern
-that can be generated with the ``advance_time`` program makes this easier to do.
+.. toctree::
+   :maxdepth: 2
+   :caption: Contents
```
Comment (hkershaw-brown) [inline observations/obs_converters/README.rst:6]: remove toc tree for individual pages.
Use toc tree at index.rst

File: guide/adding-your-observations-to-dart.rst
Comment (hkershaw-brown) [inline guide/adding-your-observations-to-dart.rst:1]: not sure it is a good idea to delete this.

File: guide/available-observation-converters.rst [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+
+First, you should understand that DART already supports a tremendous variety of
+observations. To fully support an observation means that the observation can be
+converted from its native format to the DART observation sequence format and
+that the observation forward operator is already implemented. Keep in mind that
+forward operators are not specific to any one model.
+
+The observation converters are in the *observations/obs_converter* directory and
+you should look there for the documentation describing which converters are
+available.
```
Comment (hkershaw-brown) [inline guide/available-observation-converters.rst:12]: you should look to the doc page for the documentation.

File: observations/obs_converters/README.rst [hunk 3]
```diff
@@ -394,49 +78,187 @@ converters) include:
...
+For generating synthetic observations, see the `create_obs_sequence <../../assimilation_code/programs/create_obs_sequence/create_obs_sequence.html>`__ program documentation. You can also generate observation files based on text input. See the `text_to_obs <text/text_to_obs.html>`__ program documentation and even_sphere. Or, for simulating a large complex observing system, you can use the DART library routines in a Fortran program to compute the observation information and have the DART routines write the output file.
+
+There are a couple of notable utilities of note:
+
+-  `even_sphere <even_sphere/README.html>`__ - a utility for generating a text file of evenly-spaced observation locations that can then be used in a perfect model experiment.
+-  `obs_error <obs_error/README.html>`__ - modules that specify observation errors based on what is used by ECMWF and NCEP
+
+See the `perfect_model <../../assimilation_code/programs/perfect_model_obs/perfect_model_obs.html>`__ program documentation on how to run a model with a set of observations that have only locations, types, and times and have the forward operators compute the observation values.
+
+Adding Your Observations to DART
```
Comment (hkershaw-brown) [inline observations/obs_converters/README.rst:101]: duplicated this documenation

File: observations/obs_converters/README.rst
Comment (hkershaw-brown) [inline observations/obs_converters/README.rst:1]: this file is hard to review, seems like lots of changes in format but not content.
This makes the content changes hard to find. 
I think there might have been some auto formatting from an IDE.

File: observations/obs_converters/new_converter.sh
Comment (hkershaw-brown) [inline observations/obs_converters/new_converter.sh:1]: the new_converter script is not running correctly:

```
[hkershaw:obs_converters](auto_obs_convert) > ./new_converter.sh boggy txt
boggy created successfully!
To get started,
  cd boggy/work
[hkershaw:obs_converters](auto_obs_convert) > cd boggy 
[hkershaw:boggy](auto_obs_convert) > ls
boggy_to_obs.f90   boggy_to_obs.rst   boggy_to_obs.rst-e data               readme.rst         readme.rst-e       shell_scripts      work
```

creating junk files:

'boggy_to_obs.rst-e', 'readme.rst-e'

File: observations/obs_converters/template/threed_sphere_csv_converter_mod.f90 [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+!
+
+program threed_csv_converter_mod
+
+!> title = "Generalized 3D Sphere CSV Observation Converter"
+!> institution = "NCAR" ;
+!> source = "NCAR/DAReS" ;
+!> comment = "Generalized converter for 3D sphere data from CSV files" ;
+!> references = "http://www.image.ucar.edu/DAReS/DART/DART_download" ;
+!> dataset_title = "Generalized 3D Sphere CSV Data" ;
```
Comment (hkershaw-brown) [inline observations/obs_converters/template/threed_sphere_csv_converter_mod.f90:13]: is this some auto created doxygen documentation? 
```suggestion
```

File: observations/obs_converters/template/threed_sphere_csv_converter_mod.f90 [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+use obs_utilities_mod, only : add_obs_to_seq, create_3d_obs
+
+use obs_kind_mod, only : YOUR_OBS_KIND, get_kind_index
+
+implicit none
+
+! version controlled file description for error handling, do not edit
+character(len=*), parameter :: source   = & '$URL$'
+character(len=*), parameter :: revision = '$Revision$'
+character(len=*), parameter :: revdate  = '$Date$'
```
Comment (hkershaw-brown) [inline observations/obs_converters/template/threed_sphere_csv_converter_mod.f90:44]: nope, this is svn info
```suggestion
```

File: observations/obs_converters/template/threed_sphere_csv_converter_mod.f90 [hunk 6]
```diff
@@ -0,0 +1,293 @@
+! DART software - Copyright UCAR. This open source software is provided
+! by UCAR, "as is", without charge, subject to all terms of use at
+! http://www.image.ucar.edu/DAReS/DART/DART_download
+!
+
+program threed_csv_converter_mod
```
Comment (hkershaw-brown) [inline observations/obs_converters/template/threed_sphere_csv_converter_mod.f90:6]: not a module, program threed_csv_converter

File: observations/obs_converters/template/work/quickbuild.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+# DART software - Copyright UCAR. This open source software is provided
+# by UCAR, "as is", without charge, subject to all terms of use at
+# http://www.image.ucar.edu/DAReS/DART/DART_download
+
+main() {
+
+export DART=$(git rev-parse --show-toplevel)
+source "$DART/build_templates/buildconvfunctions.sh"
+
+CONVERTER=$(basename "$(dirname "$PWD")")
```
Comment (hkershaw-brown) [inline observations/obs_converters/template/work/quickbuild.sh:11]: This does not support out of core builds, 
CONVERTER should be set to the CONVERTER name

**Comment by hkershaw-brown** [comment]
Thanks for this Charlotte, 
closing this for now as there is quite a bit of cleanup needed, so we're not running the github actions and doc builds, and pull request notifications as this branch is cleaned up.  
moved the code to 
https://github.com/NCAR/DART/tree/auto_obs_convert

**Pull request closed without merging**